internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -280,9 +280,17 @@ func (n *Networkd) decideHostname() (hostname string, domainname string, address
 	address = net.ParseIP("127.0.1.1")
 	hostname = fmt.Sprintf("%s-%s", "talos", strings.ReplaceAll(address.String(), ".", "-"))
 
+	// Sort interface names alphabetically so we can ensure parsing order
+	interfaceNames := make([]string, len(n.Interfaces))
```

Comment:
this should be `make([]string, 0, len(n.Interfaces))`, as otherwise it creates list of length N and appends N items

Reply:
ah yep. fixed in amended commit. thx.